### PR TITLE
refactor: getURL fn

### PR DESCRIPTION
**Pull request by Ayushsanjdev**
1. Desctructured tasks for better undefined check.
2. Create default params constants and used it in place of hardcoded params
3. clean conditional a little

Anyone can review from JS|TS group, Do review when you get time for this.

### CODE REVIEW

File: refactor/urls.js [hunk 1]
```diff
@@ -1,16 +1,19 @@
...
+function getURL(dev = false, tasks = {}, params = DEFAULT_TASKS_PARAMS) {
+    const { nextTasks = false, prevTasks = false } = tasks;
+    const { status, size } = params;
 
-    if (tasks.nextTasks) {
-        url += '?hasNext=true';
-    }
+    let url = TASKS_BASE_ENDPOINT;
+    if (dev)
+        url = `${TASKS_BASE_ENDPOINT}?status=${status}&dev=${dev}&size=${size}`;
```
Comment (satyam73) [inline refactor/urls.js]: what if status or size is undefined what will be you resultant URL will look like?

File: refactor/urls.js [hunk 2]
```diff
@@ -1,16 +1,19 @@
...
+
+    if (nextTasks) url += "?hasNext=true";
```
Comment (satyam73) [inline refactor/urls.js]: This will fail, since in query params question mark is already present at ln 13

File: refactor/urls.js [hunk 3]
```diff
@@ -1,16 +1,19 @@
...
+    if (prevTasks) url = `${TASKS_BASE_ENDPOINT}?hasPrev=true`;
```
Comment (satyam73) [inline refactor/urls.js]: In this we can consider dev and other query params also

File: refactor/urls.js (hunk 1 above)
Comment (Ayushsanjdev) [reply refactor/urls.js:13]: If you can see DEFAULT_TASKS_PARAMS, I have made it as default.

File: refactor/urls.js (hunk 2 above)
Comment (Ayushsanjdev) [reply refactor/urls.js]: oh right

**Comment by Ayushsanjdev** [comment]
@satyam73 ,
Can you please check now, I have refactored it.

**Comment by Ayushsanjdev** [comment]
@satyam73 , refactored to accept any no of params

File: refactor/urls.js [hunk 4]
```diff
@@ -1,16 +1,17 @@
-// Refactor this code
+const TASKS_BASE = "/tasks";
```
Comment (satyam73) [inline refactor/urls.js]: Why is this a constant which is not being used again ? Can we not declare url as /tasks

File: refactor/urls.js [hunk 5]
```diff
@@ -1,16 +1,17 @@
...
 
-function getURL(dev = false, tasks = {}) {
-    let url = dev
-        ? `/tasks?status=ACTIVE&dev=true&size=20`
-        : '/tasks';
+function getURL(dev = false, tasks = {}, params) {
+    const { nextTasks, prevTasks } = tasks;
```
Comment (satyam73) [inline refactor/urls.js]: What if previous and next task are undefined?

File: refactor/urls.js [hunk 6]
```diff
@@ -1,16 +1,17 @@
...
-    if (tasks.nextTasks) {
-        url += '?hasNext=true';
-    }
+    let url = TASKS_BASE;
+    const queryParams = new URLSearchParams({
+        ...params
+    });
+    const stringParams = queryParams.toString();
+
+    if (dev) url += `?${stringParams}`;
```
Comment (satyam73) [inline refactor/urls.js]: The if blocks doesn't seem to be readable

File: refactor/urls.js [hunk 7]
```diff
@@ -1,16 +1,17 @@
...
 
-    if (tasks.nextTasks) {
-        url += '?hasNext=true';
-    }
+    let url = TASKS_BASE;
+    const queryParams = new URLSearchParams({
+        ...params
+    });
+    const stringParams = queryParams.toString();
+
```
Comment (satyam73) [inline refactor/urls.js:15]: Can we please add comments to if blocks

File: refactor/urls.js [hunk 8]
```diff
@@ -1,16 +1,17 @@
...
-        : '/tasks';
+function getURL(dev = false, tasks = {}, params) {
+    const { nextTasks, prevTasks } = tasks;
 
-    if (tasks.nextTasks) {
-        url += '?hasNext=true';
-    }
+    let url = TASKS_BASE;
+    const queryParams = new URLSearchParams({
+        ...params
```
Comment (satyam73) [inline refactor/urls.js:12]: What if params are undefined then what will URLSearchParams return ?

**Comment by Ayushsanjdev** [comment]
updated @satyam73

**Pull request closed without merging**